bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEPFactory.java

```diff
@@ -97,7 +99,21 @@ public static EEP buildEEPFromTeachInERP1(ERP1Message msg) {
                     }
                 } catch (Exception e) {
                 }
+                try {
+                    EEP result = new F6_10_00_EltakoFPE1(msg);
+                    if (result.isValid()) {
```

Comment:
Do we have a more reliable chance to detect these two new EEPs? Can we distinguish FPE1 messages from FPE2 messages? I think we should at least override the validateData function of these EEPs and check if we received a 0x10 or 0x00. You could also change the check order of your EEP and F6_10_01. Your EEP should never be recognized as F6_10_01 as I check [bit 6, 3 and 2](https://github.com/openhab/openhab2-addons/blob/ecc69c76dc6fb023e56eca7b67b40e61ae6ef3d0/bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/F6_10/F6_10_01.java#L79). Currently we are no longer able to autodectect a F6_10_01 EEP (window handle).

Reply:
You're right. Sorry, I'm starting to understand your code, but don't have the full view yet. The only FPE-1 and FPE-2 documentation I have is from the web, and according to that, they only send one byte 0x00 and 0x10 for Open/Closed or vice versa, so there's no change to distinguish FPE-1 from FPE-2 messages. So that distinction should be made by configuration. I don't own a FPE-2, only FPE-1's so I have to trust on Eltako's documentation about only swapping the OPEN/CLOSED values.
If I validate the FPE-X-data by the pattern 000X0000, that should work, what do you think?
Implementation will last a few days, since I don't have experience with the evaluation of configuration options.

Reply:
I also googled a bit and could not find any way to distinguish FPE-1 from FPE-2 messages too. So I would suggest the following:
 * add the validation pattern 0b000X0000
 * just autodetect either FPE-1 or FPE-2, the other one must be manually reconfigured by choosing the right receiving EEP
I would not add any more configuration options and just keep your two EEP implementations.
Currently RPS autodetection is pure guessing. I think I have to improve it a little bit. Could you please post the raw message of your contact sensor for closed and open?

Reply:
Your post came just a little bit too late for my commit. Here are the RAW messages:
OPEN: F600XXXXXXXX20
CLOSED: F610XXXXXXXX20
X-es are the 4 byte enocean ID of the sensor.
In my current commit I added a boolean configuration option for contact channel named "inverted" which swaps OPEN/CLOSED and defaults to 'false'. Maybe this could be useful also for other contact sensors? You decide, I can switch back.

Reply:
Thanks for your changes. That channel config parameter is really useful 👍 . However could you do me favor and check this paramter in EEP A5-14, D5-00 and F6-10, as we use the same contact channel there too. Furthermore it would be nice if you could state this new option in readme.md too. Thanks a lot in advance.

Reply:
I just returned from vacation and will start the tasks during this week.